### PR TITLE
[zero] add AgChunk

### PR DESCRIPTION
**Pull request by 1SAA**
Add AgChunk for later update plan.
The unitest will be merged from the next pull request.

### CODE REVIEW

**Review by feifeibear — DISMISSED** [review]

File: colossalai/gemini/ag_chunk.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+    def __update_tensors_ptr(self) -> None:
+        # sanity check
+        assert self.is_gathered
+        assert type(self.chunk_total) == torch.Tensor
+
+        for tensor, tensor_info in self.tensors_info.items():
+            tensor.data = self.chunk_total[tensor_info.offset:tensor_info.end].view(tensor.shape)
+
+    def __update_tensor_info(self, tensor_info: TensorInfo, next_state: TensorState):
```
Comment (feifeibear) [inline colossalai/gemini/ag_chunk.py:357]: use one _

File: colossalai/gemini/ag_chunk.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+from colossalai.utils import get_current_device
+from colossalai.gemini.chunk import TensorState, STATE_TRANS, TensorInfo, ChunkFullError, \
+    free_storage, alloc_storage
+
+
+class AgChunk:
+    def __init__(self,
+                 chunk_size: int,
+                 process_group: dist.ProcessGroup,
```
Comment (ver217) [inline colossalai/gemini/ag_chunk.py:13]: It is torch pg

File: colossalai/gemini/ag_chunk.py (hunk 1 above)
Comment (1SAA) [reply colossalai/gemini/ag_chunk.py:357]: done

File: colossalai/gemini/ag_chunk.py (hunk 2 above)
Comment (1SAA) [reply colossalai/gemini/ag_chunk.py:13]: Yes. I use torch pg here. I have explicate in below coments. Is there any problem?

Comment (1SAA) [reply colossalai/gemini/ag_chunk.py]: done

**Review by feifeibear — APPROVED** [review]

**Pull request merged**